src/rebar3-nix-bootstrap

```diff
@@ -0,0 +1,219 @@
+    ok = if_compile_ports_add_pc_plugin(RequiredData).
+
+-spec bootstrap_libs(#data{}) -> ok.
+bootstrap_libs(#data{erl_libs = ErlLibs}) ->
+    io:format("Bootstrapping dependent libraries~n"),
+    Target = "_build/default/lib/",
+    Paths = string:tokens(ErlLibs, ":"),
+    CopiableFiles =
+        lists:foldl(fun(Path, Acc) ->
+                            gather_directory_contents(Path) ++ Acc
+                    end, [], Paths),
+    lists:foreach(fun (Path) ->
+                          make_symlink(Path, Target)
+                  end, CopiableFiles).
+
+-spec gather_directory_contents(string()) -> [string()].
+gather_directory_contents(Path) ->
+    {ok, Names} = file:list_dir(Path),
+    lists:map(fun (Name) ->
+                      filename:join(Path, Name)
+              end, Names).
+
+%% @doc
+%% Makes a symlink from the directory pointed at by Path to a
+%% directory of the same name in Target. So if we had a Path of
+%% `foo/bar/baz/bash` and a Target of `faz/foo/foos`, the symlink
+%% would be `faz/foo/foos/bash`.
+-spec make_symlink(string(), string()) -> ok.
+make_symlink(Path, Target) ->
+    AppName = filename:basename(Path),
+    TargetFile = filename:join(Target, fixup_app_name(AppName)),
+    filelib:ensure_dir(TargetFile),
+    ok = case filelib:is_file(TargetFile) of
+             true ->
+                 file:delete(TargetFile);
+             false ->
+                 ok
+         end,
+    io:format("Making symlink from ~s to ~s~n", [Path, TargetFile]),
+    file:make_symlink(Path, TargetFile).
+
+%% @doc
+%% This takes an app name in the standard OTP <name>-<version> format
+%% and returns just the app name. Why? because rebar is doesn't
+%% respect OTP conventions in some cases.
+-spec fixup_app_name(string()) -> string().
+fixup_app_name(FileName) ->
+    [Name, _Version] = string:tokens(FileName, "-"),
+    Name.
+
+-spec bootstrap_registry(#data{}) -> ok.
+bootstrap_registry(#data{registry_snapshot = RegistrySnapshot}) ->
+    io:format("Bootstrapping Hex Registry for Rebar~n"),
+    make_sure_registry_snapshot_exists(RegistrySnapshot),
+    filelib:ensure_dir(?HEX_REGISTRY_PATH),
+    ok = case filelib:is_file(?HEX_REGISTRY_PATH) of
+             true ->
+                 file:delete(?HEX_REGISTRY_PATH);
+             false ->
+                 ok
+         end,
+    ok = file:make_symlink(RegistrySnapshot,
+                           ?HEX_REGISTRY_PATH).
+
+-spec make_sure_registry_snapshot_exists(string()) -> ok.
+make_sure_registry_snapshot_exists(RegistrySnapshot) ->
+    case filelib:is_file(RegistrySnapshot) of
+        true ->
+            ok;
+        false ->
+            stderr("Registry snapshot (~s) does not exist!", [RegistrySnapshot])
+    end.
+
+-spec gather_required_data_from_the_environment(#data{}) -> {ok, map()}.
+gather_required_data_from_the_environment(ArgData) ->
+    {ok, ArgData#data{ version = guard_env("version")
+                     , erl_libs = os:getenv("ERL_LIBS", [])
```

Comment:
I think ERL_LIBS is defined in Erlang now (its one of the changes that seems to be causing this command some problems) so I will do that. Just remember that the makefile for this project isn't run when the code above is run. The code above is run during the build of packages that use it.